src/Schema/Column/DateTimeColumn.php

```diff
@@ -0,0 +1,226 @@
+     * inserting them into the database or after retrieving them from the database. Use empty string to use current PHP
+     * time zone.
+     */
+    protected string $phpTimezone = 'UTC';
+
+    /**
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    protected bool $shouldConvertTimezone;
+
+    /**
+     * @psalm-var non-empty-string
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    protected string $format;
+
+    /**
+     * @param mixed $value The value representing datetime or time to be typecasted to the database format.
+     * Possible values:
+     * - null;
+     * - string - it can be:
+     *     - empty string - treated as null;
+     *     - string with integer value - treated as unix timestamp;
+     *     - string with float value - treated as unix timestamp with microseconds;
+     *     - string with datetime format;
+     * - Stringable object - converted to string;
+     * - integer - treated as unix timestamp;
+     * - float - treated as unix timestamp with microseconds;
+     * - DateTimeImmutable;
+     * - DateTimeInterface;
+     * - ExpressionInterface;
+     *
+     * If the value is `string` or a `Stringable` object, it will be converted to a `DateTimeImmutable` object with
+     * the default time zone set specified in the {@see $phpTimezone} property. If the conversion fails, the original
+     * value will be returned.
+     */
+    public function dbTypecast(mixed $value): string|ExpressionInterface|null
+    {
+        /** @psalm-suppress MixedArgument, PossiblyFalseArgument */
+        return match (gettype($value)) {
+            GettypeResult::NULL => null,
+            GettypeResult::STRING => $this->dbTypecastString($value),
+            GettypeResult::INTEGER => $this->dbTypecastDateTime(DateTimeImmutable::createFromFormat('U', (string) $value)),
+            GettypeResult::DOUBLE => $this->dbTypecastDateTime(DateTimeImmutable::createFromFormat('U.u', (string) $value)),
+            GettypeResult::OBJECT => match (true) {
+                $value instanceof DateTimeImmutable => $this->dbTypecastDateTime($value),
+                $value instanceof DateTimeInterface => $this->dbTypecastDateTime(DateTimeImmutable::createFromInterface($value)),
+                $value instanceof ExpressionInterface => $value,
+                default => $this->dbTypecastString((string) $value),
+            },
+            default => throw new InvalidArgumentException('Wrong ' . gettype($value) . ' value for ' . $this->getType() . ' column.'),
+        };
+    }
+
+    /** @psalm-mutation-free */
+    public function getPhpType(): string
+    {
+        return DateTimeImmutable::class;
+    }
+
+    /**
+     * Converts the value from the database format to PHP `DateTimeImmutable` object.
+     * If the database type does not have time zone information, the time zone will be set to the current PHP time zone.
+     *
+     * @param string|null $value
```

Comment:
How can we guarantee that there will always be string or null?

Reply:
The method is for converting values retrieved from DB and it is `string` or `null`

Reply:
> The method is for converting values retrieved from DB and it is `string` or `null`

Should we change `ColumnInterface::phpTypecast()` signature to `phpTypecast(?string $value): mixed`?

Reply:
No, for `bool`, `int` and `binary` colums DB returns `bool`, `int` and `resources`